src/components/Counter.js

```diff
@@ -1,42 +1,65 @@
 import React, { Component } from "react";
-import { connect } from 'react-redux';
-import { increment, decrement } from '../actions';
+import { connect } from "react-redux";
+import { increment, decrement } from "../actions";
 
 class Counter extends Component {
-    incrementIfOdd = () => {
-        // Stretch Problem: Implement an increment function that
-        // only increments if the counter value is odd
-    };
+  incrementIfOdd = count => {
+    // Stretch Problem: Implement an increment function that
+    // only increments if the counter value is odd
+    if (count % 2 !== 0) {
+      this.props.increment(count);
+    }
+  };
+
+  incrementAsync = count => {
+    // Stretch Problem: Implement an increment function that
+    // increments after waiting for one second
 
-    incrementAsync = () => {
-        // Stretch Problem: Implement an increment function that
-        // increments after waiting for one second
-    };
+    //idk it seemed like they would want us to do this with a promise since we practiced that last week
 
-    render() {
-        // Fill in the two button onClick methods
-        // Upon clicking these buttons, the count
-        // should decrement or increment accordingly
-        return (
-            <p>
-                Clicked: {this.props.count} times
-                <button onClick={() => {/* Fill me in */ }}>
-                    +
-                </button>
-                <button onClick={() => {/* Fill me in */ }}>
-                    -
-                </button>
-                 {/* Uncomment these button tags if you got
+    new Promise(resolve => {
+      setTimeout(() => {
+        resolve(count);
+      }, 1000);
+    }).then(res => {
+      return this.props.increment(res);
+    });
+  };
+
+  render() {
+    // Fill in the two button onClick methods
+    // Upon clicking these buttons, the count
+    // should decrement or increment accordingly
+    return (
+      <p>
+        Clicked: {this.props.count} times
+        <button
+          onClick={() => {
+            /* Fill me in */
+            this.props.increment(this.props.count);
```

Comment:
```suggestion
            this.props.increment();
```
You're increment and decrement can just fire off without needing to pass anything